src/Workspaces/Core/Portable/Workspace/Workspace_Events.cs

```diff
@@ -160,7 +160,7 @@ protected Task RaiseDocumentClosedEventAsync(Document document)
         /// An event that is fired when the active context document associated with a buffer 
         /// changes.
         /// </summary>
-        internal event EventHandler<DocumentEventArgs> DocumentActiveContextChanged
+        public event EventHandler<DocumentEventArgs> DocumentActiveContextChanged
```

Comment:
> it is the same information that is sent along with the other public Document events (such as DocumentOpened and DocumentClosed)

With DocumentOpened or Closed there aren't two different documents to be talking about :)

> Any risk that would exist here also exists there,

DocumentContextChanged is different because there's the Document you used to be talking about, and the Document you're now talking about.

> and modifying DocumentEventArgs to no longer expose the Document would be a breaking change at this point.

We don't need to do that.  We could have, for example, DocumentIdEventArgs.

> DocumentClosed is for the closed document and DocumentOpen is for the opened document

I would certainly hope so :D